components/dashboard/src/data/prebuilds/prebuild-logs-emitter.ts

```diff
@@ -23,19 +32,55 @@ export function usePrebuildLogsEmitter(prebuildId: string, taskId: string) {
     useEffect(() => {
         const controller = new AbortController();
         const watch = async () => {
+            if (taskId === null) {
+                return;
+            }
             if (!prebuildId || !taskId) {
                 setIsLoading(false);
                 return;
             }
+
             let dispose: () => void | undefined;
             controller.signal.addEventListener("abort", () => {
                 dispose?.();
             });
-            const prebuild = await prebuildClient.getPrebuild({ prebuildId });
-            const task = prebuild.prebuild?.status?.taskLogs?.find((log) => log.taskId === taskId);
+            const { prebuild } = await prebuildClient.getPrebuild({ prebuildId });
+            if (taskId === "image-build" && prebuild?.workspaceId) {
+                const workspace = await workspaceClient.getWorkspace({ workspaceId: prebuild.workspaceId });
+                if (workspace.workspace?.status?.phase?.name === WorkspacePhase_Phase.IMAGEBUILD) {
+                    await pRetry(
+                        async () => {
+                            await getGitpodService().server.watchWorkspaceImageBuildLogs(prebuild.workspaceId);
```

Comment:
I'm totally fine to have it with old APIs, but if we want to continue grpc migration, maybe worth do it in new API: add endpoint (url) in server and respond the url in papi.

And as image build's log is only accessible when it's in image build phase, I'm not sure where we want to go: If we don't plan to hide image build tab after this phase, we need to store the data in this tab

Reply:
We decided to introduce an image build log URL which the client can fetch exactly the same way as is done with other tasks.